TransIT.BLL/Services/ImplementedServices/ActionTypeService.cs

```diff
@@ -15,79 +16,82 @@ namespace TransIT.BLL.Services.ImplementedServices
     /// </summary>
     /// <see cref="IActionTypeService"/>
-    public class ActionTypeService : CrudService<ActionType>, IActionTypeService
+    public class ActionTypeService : IActionTypeService
     {
+        private readonly IUnitOfWork _unitOfWork;
+        private readonly IMapper _mapper;
+
         /// <summary>
         /// Ctor
         /// </summary>
         /// <param name="unitOfWork">Unit of work pattern</param>
-        /// <param name="logger">Log on error</param>
-        /// <param name="repository">CRUD operations on entity</param>
-        /// <see cref="CrudService{TEntity}"/>
-        public ActionTypeService(
-            IUnitOfWork unitOfWork,
-            ILogger<CrudService<ActionType>> logger,
-            IActionTypeRepository repository) : base(unitOfWork, logger, repository) { }
+        /// <param name="mapper"></param>
+        public ActionTypeService(IUnitOfWork unitOfWork, IMapper mapper)
+        {
+            _unitOfWork = unitOfWork;
+            _mapper = mapper;
+        }
+
+        public async Task<ActionTypeDTO> GetAsync(int id)
+        {
+            return _mapper.Map<ActionTypeDTO>(await _unitOfWork.ActionTypeRepository.GetByIdAsync(id));
+        }
+
+        public async Task<IEnumerable<ActionTypeDTO>> GetRangeAsync(uint offset, uint amount)
+        {
+            var entities = await _unitOfWork.ActionTypeRepository.GetRangeAsync(offset, amount);
+            return _mapper.Map<IEnumerable<ActionTypeDTO>>(entities);
+        }
+
+        public async Task<IEnumerable<ActionTypeDTO>> SearchAsync(string search)
+        {
+            var actionTypes = await _unitOfWork.ActionTypeRepository.SearchExpressionAsync(
+           search
+               .Split(new[] { ' ', ',', '.' }, StringSplitOptions.RemoveEmptyEntries)
+               .Select(x => x.Trim().ToUpperInvariant())
+           );
+
+            return _mapper.Map<IEnumerable<ActionTypeDTO>>(await actionTypes.ToListAsync());
+        }
         
-        public async override Task<ActionType> UpdateAsync(ActionType model)
+        public async Task<ActionTypeDTO> CreateAsync(ActionTypeDTO dto)
         {
-            try
-            {
-                var newModel = await GetAsync(model.Id);
-                if (newModel.IsFixed)
-                {
-                    throw new ConstraintException("Current state can not be edited");
-                }
-                if (model.IsFixed)
-                {
-                    throw new ArgumentException("Incorrect model");
-                }
-                newModel.Name = model.Name;
+            ActionType model = _mapper.Map<ActionType>(dto);
 
-                _repository.Update(newModel);
-                await _unitOfWork.SaveAsync();
-                return newModel;
-            }
-            catch (DbUpdateException e)
+            await _unitOfWork.ActionTypeRepository.AddAsync(model);
+            await _unitOfWork.SaveAsync();
+            return dto;
```

Comment:
should be returned dto mapped from the model with ceated id or other data, check all such cases

`return model.MapTo(dto)`

Reply:
This way is ok?
` var model = _mapper.Map<ActionType>(dto);

            model = await _unitOfWork.ActionTypeRepository.AddAsync(model);
            await _unitOfWork.SaveAsync();
            return _mapper.Map<ActionTypeDTO>(model);`